### PR TITLE
Ui alerts

### PR DESCRIPTION
**Pull request by hansonchun**
updates to alerts 
- updated ui
- mark as read
- delete
- color coded toasts
- inbox badge

### CODE REVIEW

**Comment by mike-mh** [comment]
2,321 insertions... 492 deletions...

Dear god what a monster pull request haha!

File: dch/public/app/components/navbar/canary-alerts/canary-alerts-service.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          alert_text: 'Device has connected to network',
+          read: false
+        },
+        {
+          timestamp: testDate.toString(),
+          watch_id: '7982347af7982',
+          alert_type: 'Connection',
+          alert_text: 'Device has connected to network',
+          read: false
+        },
```
Comment (mike-mh) [inline dch/public/app/components/navbar/canary-alerts/canary-alerts-service.js]: We should remove these examples. We don't need them any more.

File: dch/public/app/components/navbar/canary-alerts/canary-alerts-service.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  function CanaryAlertService($http) {
+    var canaryAlertService = {
+      getWatchAlerts: getWatchAlerts,
+      regiterCallback: regiterCallback,
+      deleteWatchAlert: deleteWatchAlert
+    };
+
+    var ALERT_STREAM_PATH = 'api/alert-sse';
+    var ALERT_EVENT = 'alert-event';
+    var GET_ALERTS_PATH = '/get-alerts';
```
Comment (mike-mh) [inline dch/public/app/components/navbar/canary-alerts/canary-alerts-service.js]: Has this been implemented? We should be able to populate the inbox with messages already stored in the DB using this api

File: dch/mqtt-subscriber/mqtt-subscriber.js [hunk 3]
```diff
@@ -96,16 +122,153 @@ dcappClient.on(MQTT_MESSAGE_EVENT, function (topic, message) {
...
+  var dateInMilliseconds = currentDate.getTime();
+
+  var disconnectionAlert =
+    {
+      timestamp: dateInMilliseconds,
+      watch_id: uuid,
+      alert_type: 'CONNECTION',
+      alert_text: 'Device has disconnected from DCH.'
+    };
+
```
Comment (mike-mh) [inline dch/mqtt-subscriber/mqtt-subscriber.js:263]: Caught this bug last night. Need to put this object into the 'createAlert' function in order to broadcast it to DCGUI

File: dch/mqtt-subscriber/mqtt-subscriber.js [hunk 4]
```diff
@@ -96,16 +122,153 @@ dcappClient.on(MQTT_MESSAGE_EVENT, function (topic, message) {
...
+  AlertData.create(disconnectionAlert, function(err, data) {
+    if (err) {
+      console.log('There was an error inserting ' + data + ' into the database');
+    } else {
+      console.log(data.toString() + ' saved to database');
+    }
+  });
+
```
Comment (mike-mh) [inline dch/mqtt-subscriber/mqtt-subscriber.js:265]: Replace the code above with  'createAlert' function call

**Comment by mike-mh** [comment]
Just went over the submission.

Over all it looks pretty good. There is quite a bit of dead code though but at this point it isn't much of a priority.

Take care of the two issues mentined earlier, implementing the 'get-alerts' API call in the DCGUI and populating the inbox and remove the example messages and we should be good.

File: dch/public/app/components/navbar/canary-alerts/canary-alerts-service.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+     *
+     * @return {object} - The watch alerts object
+     */
+    function getWatchAlerts() {
+      return watchAlerts;
+    }
+
+    function deleteWatchAlert(alert) {
+      $http.get(DELETE_ALERT_PATH + '/:' + alert.mongo_id)
+    }
```
Comment (mike-mh) [inline dch/public/app/components/navbar/canary-alerts/canary-alerts-service.js:79]: Does this work? Shouldn't it be '/' not '/:' ?

**Comment by mike-mh** [comment]
Just patched this guy up. We should be good to go here.

@CarlosBalladares 

QA please :)

**Comment by mike-mh** [comment]
Disregard... bug found

**Comment by mike-mh** [comment]
Bug squished. Commence the Q/A!

**Comment by mike-mh** [comment]
Things to test:
- Alerts are available for:
  - when a watch is idle for five minutes
  - when a watch connects
  - when a watch disconnects
  - when a watch experiences an acceleration spike
- When an alert occurs:
  - a 'toast' should appear with the right information
  - the alert should persist in the Inbox with the right information
  - The 'unread' number should increase by 1
- Using the alert inbox
  - When an alert is clicked as 'read' it should move to the 'read' section
  - When an alert is deleted it should be removed
  - When the page is refreshed, the total unread should include those retrieved from the server
  - When a message is deleted, it does not reappear in the inbox upon refresh

**Comment by CarlosBalladares** [comment]
Everything is working ok.

**Comment by CarlosBalladares** [comment]
Now it's working after merge issues, proceeding to merge.

**Pull request merged**